mono/mini/interp/interp.c

```diff
@@ -3707,6 +3701,86 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 			ip += 2;
 			MINT_IN_BREAK;
 		}
+		MINT_IN_CASE(MINT_NEWOBJ_ARRAY) {
+			MonoClass *newobj_class;
+			InterpMethod *imethod;
+			MonoMethodSignature *csig;
+			guint32 token = * (guint16 *)(ip + 1);
+
+			imethod = (InterpMethod*) rtm->data_items [token];
+			csig = mono_method_signature (imethod->method);
+			newobj_class = imethod->method->klass;
+
+			sp -= csig->param_count;
+			sp->data.p = ves_array_create (rtm->domain, newobj_class, csig, sp, error);
+			if (!mono_error_ok (error))
+				THROW_EX (mono_error_convert_to_exception (error), ip);
+
+			++sp;
+			ip += 2;
+			MINT_IN_BREAK;
+		}
+		MINT_IN_CASE(MINT_NEWOBJ_FAST)
+		MINT_IN_CASE(MINT_NEWOBJ_VT_FAST)
+		MINT_IN_CASE(MINT_NEWOBJ_VTST_FAST) {
+			guint16 param_count;
+			gboolean vt = *ip != MINT_NEWOBJ_FAST;
+			stackval valuetype_this;
+
+			frame->ip = ip;
+
+			child_frame.imethod = (InterpMethod*) rtm->data_items [*(guint16*)(ip + 1)];
+			param_count = *(guint16*)(ip + 2);
+
+			if (param_count) {
+				sp -= param_count;
+				memmove (sp + 1, sp, param_count * sizeof (stackval));
+			}
+			child_frame.stack_args = sp;
+
+			if (vt) {
+				gboolean vtst = *ip == MINT_NEWOBJ_VTST_FAST;
+				memset (&valuetype_this, 0, sizeof (stackval));
+				if (vtst) {
+					sp->data.p = vt_sp;
+					valuetype_this.data.p = vt_sp;
+				} else {
+					sp->data.p = &valuetype_this;
+				}
+			} else {
+				MonoVTable *vtable = (MonoVTable*) rtm->data_items [*(guint16*)(ip + 3)];
+				if (G_UNLIKELY (!vtable->initialized)) {
```

Comment:
@kumpera I don't think it is worth it for now. It would probably lead to a perf gain significantly lower than 1%. (based on a similar improvement we did with inlining the class instance size)